### PR TITLE
Feature/prplmesh bpl rdkb adaptations

### PR DESCRIPTION
**Pull request by morantr**
Add changes which are needed to compile and run prplmesh on rdkb.

### CODE REVIEW

File: framework/platform/bpl/uci/db/prplmesh_db [hunk 1]
```diff
@@ -1,4 +1,4 @@
-config beerocks 'config'
```
Comment (tomereli) [inline framework/platform/bpl/uci/db/prplmesh_db]: This file should be deleted. It should be part of the feed.

**Review by tomereli — CHANGES_REQUESTED** [review]
This PR is completely wrong - prplmesh UCI configuration should NOT be part of the prplMesh repo.
Commits 2 and 3 are OK as a separate PR.

File: framework/platform/bpl/uci/db/prplmesh_db (hunk 1 above)
Comment (vitalybu) [reply framework/platform/bpl/uci/db/prplmesh_db]: I think it's fine to have this as a "template" in the repo.
The UCI BPL reads various configurations, and this file is the only place that shows how the DB should look.
It can even help bootstrap DBs for other UCI based platforms.

Comment (arnout) [reply framework/platform/bpl/uci/db/prplmesh_db]: That's not really how UCI is conceived. UCI is supposed to be empty if defaults are used (i.e., if a config option is missing, it should just work - that's why there are `disabled` options instead of `enabled` options). UCI configs which are absolutely required should be generated by a script in /etc/uci-defaults.

In upstream openwrt, the extra scripts (uci defaults, init, ...) are always part of the openwrt integration, not part of the source repository, even for openwrt-specific packages (e.g. odhcpd). However, I'm not sure that I agree with openwrt's approach there: it is much easier to make changes to uci options when the uci defaults file is part of the source repository. And the package integration can still override it.

Bottom line: I agree with @vitalybu that it's better to put UCI defaults in the prplMesh repository (but not install it, that should be done explicitly by the package logic). However, it should be done as a uci-defaults script instead of a static file.

I do realize that this poses a problem for rdkb, because it doesn't have uci and prplmesh is still using it. But that's something I think should be fixed in the rdkb integration itself.

Comment (tomereli) [reply framework/platform/bpl/uci/db/prplmesh_db]: Actually at least the Intel RDKB has uci, this is why we have the same uci file for both rdkb and UGW.

So if this is the case then I agree to add the defaults here but it should be supplemented with PRs in UGW and prplwrt feeds which will install it to /etc/uci-defaults and remove it from the feed files, so eventually we will need to maintain a single file.

File: framework/platform/bpl/uci/db/prplmesh_db [hunk 2]
```diff
@@ -16,5 +16,5 @@ config beerocks 'config'
         option key_passphrase 'test_passphrase'
         option mem_only_psk '0'
         option backhaul_band 'auto'
-
-
+        option certification_mode '0'
```
Comment (tomereli) [inline framework/platform/bpl/uci/db/prplmesh_db]: certification_mode should be '1'

Comment (tomereli) [reply framework/platform/bpl/uci/db/prplmesh_db]: Please make sure this is aligned to latest UGW. Note that I changed latest UGW.

File: framework/platform/bpl/CMakeLists.txt [hunk 3]
```diff
@@ -73,6 +74,10 @@ elseif (TARGET_PLATFORM STREQUAL "rdkb")
 
     list(APPEND BPL_LIBS nl-3 nl-route-3 bridge dl swpal uci ubox ubus slibc)
 
+    # Install the UCI DB files
```
Comment (tomereli) [inline framework/platform/bpl/CMakeLists.txt:77]: Should be installed to `${CMAKE_INSTALL_PREFIX}/etc/uci-defaults/` - @arnout please confirm.

**Review by tomereli — CHANGES_REQUESTED** [review]
Still need changes. Also need @arnout to approve.

File: framework/platform/bpl/CMakeLists.txt (hunk 3 above)
Comment (arnout) [reply framework/platform/bpl/CMakeLists.txt:77]: It should be just plain `/etc` - CMAKE_INSTALL_PREFIX defaults to /usr (or /usr/local, I'm not sure) and is overridden by us to /opt/prplmesh (or /opt/beerocks, I'm not sure). We need plain and simple /etc. Note that the DESTDIR is still prepended to that, i.e. if you install with `make install DESTDIR=/path/to/openwrt/staging` (which is what openwrt/ugw does), it goes to /path/to/openwrt/staging/etc which is exactly where we want it.

*However* /etc/uci-defaults if for *scripts* which *generate* the initial database, based on runtime information (e.g. actual network interfaces on the board). So if we put it there, it should be:

- executable;
- a script that checks if the uci config already exists, and exit if not;
- a filename starting with numbers, e.g. 00_prplmesh.

Comment (tomereli) [reply framework/platform/bpl/CMakeLists.txt:77]: Decision is to leave it as is.

File: framework/platform/bpl/uci/db/prplmesh_db (hunk 2 above)
Comment (tomereli) [reply framework/platform/bpl/uci/db/prplmesh_db]: https://git.prpl.dev/prplmesh/iwlwav/blob/iwlwav-19.07/prplmesh/files/etc/config/prplmesh

**Review by tomereli — APPROVED** [review]

File: common/beerocks/scripts/prplmesh_utils.sh.in [hunk 4]
```diff
@@ -142,16 +142,19 @@ start_function() {
     [ "@BTL_TYPE@" = "LOCAL_BUS" ] && prplmesh_framework_init
     case "$PRPLMESH_MODE" in
         CA | ca)
-            prplmesh_platform_db_init "Multi-AP-Controller-and-Agent"
+            [ "@BWL_TYPE@" = "DUMMY" -a "$PLATFORM_INIT" = "true" ] && \
```
Comment (arnout) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: It would make more sense to have this condition directly in the prplmesh_platform_db_init function (nitpick).

Comment (arnout) [inline common/beerocks/scripts/prplmesh_utils.sh.in]: The `"$PLATFORM_INIT" = "true"` part will cause the platform_db to not be initialized when we run our tests, because test_gw_repeater.sh skips platform initialisation. It's also not clear to me why you don't want to do this when PLATFORM_INIT is false - PLATFORM_INIT is about setting up the virtual networks for dummy operation...

**Review by arnout — APPROVED** [review]
OK to merge after fixing the test.

File: common/beerocks/scripts/prplmesh_utils.sh.in (hunk 4 above)
Comment (morantr) [reply common/beerocks/scripts/prplmesh_utils.sh.in]: I dropped this commit
